### PR TITLE
[Pages] fix incorrect instructions for managing multiple environments

### LINKED ISSUE #1820: Instructions for Private Packages for NPM Don't Work as Expected
Referencing the previous issue, which was closed. https://github.com/cloudflare/cloudflare-docs/pull/1334

## Expected Behavior
By using a `preinstall` script you should be able to copy an `.npmrc.pages` over the existing `.npmrc` file in order to authenticate to private packages on GitHub.

## Actual Behavior
npm does not reload the `.npmrc` settings after the `preinstall` script so copying the `.npmrc.pages` file to `.npmrc` has no effect.

### Replicating
To avoid any issues with permissions issues with tokens and specific environments I've replicated this issue in a new environment with a validated token. 

I've done three different tests do demonstrate the issue. Each test used the same `TOKEN_FOR_GITHUB` environment variable.

### Test One ✅
Directly commit the `.npmrc` with the `${TOKEN_FOR_GITHUB}` placeholder to the repository and deploy.

`.npmrc`:
```
@tailwind:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=${TOKEN_FOR_GITHUB}
```

This test successfully installed npm packages from a private repository using the token.

This validates that the token is set correctly and that the configuration of the `.npmrc` file is correct.

Unfortunately this is not a tenable option, since it breaks all local development unless the environment has a `TOKEN_FOR_GITHUB` set.

### Test Two ❌
Commit an  `.npmrc` with only the registry information, and create a `.npmrc.pages` file that has the `${TOKEN_FOR_GITHUB}`.

Deploy these using the instructions provided `is-pages.js` and `prepublish` script from the documentation, along with using cat to dump the contents of the file.

`.npmrc`:
```
@tailwind:registry=https://npm.pkg.github.com
```

`.npmrc.pages`:
```
@tailwind:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=${TOKEN_FOR_GITHUB}
```

`is-pages.js`:
```js
if (!process.env.CF_PAGES) process.exit(1);
```

`package.json`:
```json
  "preinstall": "node is-pages && mv .npmrc.pages .npmrc || echo \"Skipping Cloudflare Pages\" && cat .npmrc",
```

This failed to build because npm could not authenticate with Github. The contents of the `.npmrc` were correct and matched the contents of the checked in `.npmrc.pages`.

This shows that npm is not respecting changes to `.npmrc` after it has started the install process.

### Test Three ❌
Do not create an `.npmrc` and create a `.npmrc.pages` file that has the `${TOKEN_FOR_GITHUB}`.

Deploy these using the instructions provided `is-pages.js` and `prepublish` script from the documentation, along with using cat to dump the contents of the file.

`.npmrc.pages`:
```
@tailwind:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=${TOKEN_FOR_GITHUB}
```

`is-pages.js`:
```js
if (!process.env.CF_PAGES) process.exit(1);
```

`package.json`:
```json
  "preinstall": "node is-pages && mv .npmrc.pages .npmrc || echo \"Skipping Cloudflare Pages\" && cat .npmrc",
```

This failed to build because npm could not authenticate with Github. The contents of the `.npmrc` were correct and matched the contents of the checked in `.npmrc.pages`.

This validates that there is no difference between having an existing `.npmrc` in the repository and having none.

## Section that requires update
The "Managing Multiple Environments" section of the documentation should be replaced with a disclaimer that this is not possible

https://developers.cloudflare.com/pages/how-to/npm-private-registry#managing-multiple-environments

This should be added to the "Known Issues" section.

https://developers.cloudflare.com/pages/platform/known-issues

## Additional information
I am able to replicate this consistently and reliably, and I believe the extend of my testing has covered all possibilities. 

I would be happy to provide access to a private github repository with private packages that is experiencing this issue if further evidence is needed.

Thanks.

### PR DESCRIPTION
**Pull request by 10p-freddo**
Fixes #1820

### CODE REVIEW

**Comment by gmemstr** [comment]
@deadlypants1973 Any news on getting this reviewed/merged? I imagine this is a stumbling block for quite a few people exploring Cloudflare Pages for use (myself included)!

File: products/pages/src/content/how-to/npm-private-registry.md [hunk 1]
```diff
@@ -78,21 +78,4 @@ In the event that your local development no longer works with your new `.npmrc`
...
-    ```js
-    // package.json
-    {
-      "scripts": {
-        "preinstall": "node is-pages && mv .npmrc.pages .npmrc || echo \"Not Pages\"",
-        // your existing scripts
-      }
-    }
-    ```
+3. In your Pages project's **Settings** > **Environment variables**, add a new [environment variable](/platform/build-configuration#environment-variables) named [`NPM_CONFIG_USERCONFIG`](https://docs.npmjs.com/cli/v6/using-npm/config#npmrc-files) and set its value to `/opt/buildhome/repo/.npmrc.pages`. If your `.npmrc.pages` file is not in your project's root directory, adjust this path accordingly.
```
Comment (deadlypants1973) [inline products/pages/src/content/how-to/npm-private-registry.md]:
```suggestion
3. Go to your Pages project > **Settings** > **Environment variables**, add a new [environment variable](/platform/build-configuration#environment-variables) named [`NPM_CONFIG_USERCONFIG`](https://docs.npmjs.com/cli/v6/using-npm/config#npmrc-files) and set its value to `/opt/buildhome/repo/.npmrc.pages`. If your `.npmrc.pages` file is not in your project's root directory, adjust this path accordingly.
```
```suggestion
3. In your Pages project's **Settings** > **Environment variables**, add a new [environment variable](/platform/build-configuration#environment-variables) named [`NPM_CONFIG_USERCONFIG`](https://docs.npmjs.com/cli/v6/using-npm/config#npmrc-files) and set its value to `/opt/buildhome/repo/.npmrc.pages`. If your `.npmrc.pages` file is not in your project's root directory, adjust this path accordingly.
```

Comment (deadlypants1973) [inline products/pages/src/content/how-to/npm-private-registry.md]:
```suggestion
3. Go to your Pages project > **Settings** > **Environment variables**, add a new [environment variable](/platform/build-configuration#environment-variables) named [`NPM_CONFIG_USERCONFIG`](https://docs.npmjs.com/cli/v6/using-npm/config#npmrc-files) and set its value to `/opt/buildhome/repo/.npmrc.pages`. If your `.npmrc.pages` file is not in your project's root directory, adjust this path accordingly.
```
```suggestion
3. In your Pages project's **Settings** > **Environment variables**, add a new [environment variable](/platform/build-configuration#environment-variables) named [`NPM_CONFIG_USERCONFIG`](https://docs.npmjs.com/cli/v6/using-npm/config#npmrc-files) and set its value to `/opt/buildhome/repo/.npmrc.pages`. If your `.npmrc.pages` file is not in your project's root directory, adjust this path accordingly.
```

**Review by deadlypants1973 — APPROVED** [review]
Approving this for style guide review, but need a DA to check as well.

**Review by WalshyDev — APPROVED** [review]

**Review by deadlypants1973 — APPROVED** [review]

**Pull request closed without merging**